api/user.go

```diff
@@ -814,7 +817,22 @@ func sendPasswordReset(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if sent, err := app.SendPasswordReset(email, c.GetSiteURL()); err != nil {
+	if user, err = app.GetUserByEmail(email); err != nil {
+		return
+	}
+
+	if user.AuthData != nil && len(*user.AuthData) != 0 {
+		c.Err = model.NewLocAppError("SendPasswordReset", "api.user.send_password_reset.sso.app_error", nil, "userId="+user.Id)
+		return
+	}
+
+	var recovery *model.PasswordRecovery
+	if recovery, err = app.CreatePasswordRecovery(user.Id); err != nil {
+		c.Err = err
+		return
+	}
+
+	if sent, err := app.SendPasswordReset(email, recovery, user.Locale, c.GetSiteURL()); err != nil {
```

Comment:
@cpanato Can you move all the above code back into the `app` package? You can still have a separate function for sending the email in `app/email.go` but that can be called `SendPasswordResetEmail()` and used from within `app/user.go` where this code originally was